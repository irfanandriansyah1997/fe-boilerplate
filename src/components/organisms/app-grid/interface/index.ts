import { FC } from 'react';

import { ICellProps, IGridProps } from '../../../molecules/grid/interface';

/**
 * App Grid Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.30
 */
export interface IAppGridProps extends IGridProps {
  Cell: FC<ICellProps>;
  onForceRerender(): void;
  onUpdateGrid(): void;
}
