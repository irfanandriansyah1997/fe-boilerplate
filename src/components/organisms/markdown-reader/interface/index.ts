/**
 * Markdown Content Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.15
 */
export interface IMarkdownContentProps {
  content: string;
}

/**
 * Markdown Hooks Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.15
 */
export interface IMarkdownHooks {
  content?: string;
  status: number;
  url: string;
}

/**
 * Markdown Reader Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.15
 */
export interface IMarkdownReaderProps {
  markdownUrl: string;
}
