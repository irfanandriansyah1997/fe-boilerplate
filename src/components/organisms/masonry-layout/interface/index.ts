import { FC } from 'react';

/**
 * Grouping Masonry Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.15
 */
export interface IGroupingMasonry {
  item: IMasonryItem[];
  key: string;
}

/**
 * Masonry Hooks Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.24
 */
export interface IMasonryHooks {
  gridCount: number;
  item: IGroupingMasonry[];
}

/**
 * Masonry Item Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.15
 */
export interface IMasonryItem {
  item: Record<string, any>;
  key: string;
}

/**
 * Masonry Layout Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.15
 */
export interface IMasonryLayout {
  component: FC<any>;
  item: IMasonryItem[];
  maxGrid: number;
}
