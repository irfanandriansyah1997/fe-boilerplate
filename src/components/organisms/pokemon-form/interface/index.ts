/**
 * Pokemon Form Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.11.19
 */
export interface IPokemonFormProps {
  initialPokemonName?: string;
  onSubmit(pokemonName: string): void;
  pokemonName: string;
}

/**
 * Pokemon Option Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.11.19
 */
export interface IPokemonOptionProps {
  onClick(pokemonName: string): void;
  pokemonName: string;
}
