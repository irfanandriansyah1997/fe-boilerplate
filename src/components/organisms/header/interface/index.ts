import { IMenuItem } from '../../../../interface/component';

/**
 * Header Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.14
 */
export interface IHeaderProps {
  menu: IMenuItem[];
}
