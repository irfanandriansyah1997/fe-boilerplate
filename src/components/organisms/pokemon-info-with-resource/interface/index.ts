import { IPokemonCachePayload } from '../../../../section/suspense-api/suspense-image/interface';
import { IPokemonCardTypeEnum } from '../../../molecules/pokemon-card/interface';

/**
 * Pokemon Info With Resource Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.11.19
 */
export interface IPokemonInfoWithResourceProps {
  resource: IPokemonCachePayload;
  type?: IPokemonCardTypeEnum;
}
