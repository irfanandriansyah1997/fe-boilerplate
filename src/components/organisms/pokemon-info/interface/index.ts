import { IResourcePayload } from '../../../../interface/general';
import { IPokemonDetail } from '../../../../interface/pokemon';

/**
 * Pokemon Info Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.11.19
 */
export interface IPokemonInfoProps {
  lazyLoadImages?: boolean;
  resource: IResourcePayload<IPokemonDetail>;
}
