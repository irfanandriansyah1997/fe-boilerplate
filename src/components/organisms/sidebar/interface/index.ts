import { IMenuItem } from '../../../../interface/component';

/**
 * Sidebar Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.10.14
 */
export interface ISidebarProps {
  menu: IMenuItem[];
  subtitle: string;
  title: string;
}
