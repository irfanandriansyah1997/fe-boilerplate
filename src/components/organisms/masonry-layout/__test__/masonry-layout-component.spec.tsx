import { act, render } from '@testing-library/react';
import { FC } from 'react';

import { simulateResize } from '../../../../helper';
import { IMasonryItem } from '../interface';
import MasonryLayout from '..';

/**
 * Sample Component
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.11.05
 */
const SampleComponent: FC = () => (
  <div data-testid="masonry-item">Hello World</div>
);

const MOCK_ITEM: IMasonryItem[] = Array.from({
  length: 8
}).map(
  (_, index): IMasonryItem => ({
    item: {
      name: `name-${index}`,
      no: index
    },
    key: `name-${index}`
  })
);

describe(`Testing Masonry Layout Componet`, () => {
  it(`Snapshot Testing`, async () => {
    const { findByTestId } = render(
      <MasonryLayout component={SampleComponent} item={MOCK_ITEM} maxGrid={4} />
    );

    await act(simulateResize(1366));
    expect(await findByTestId(/masonry-container/)).toBeInTheDocument();
    expect(await findByTestId(/masonry-container/)).toMatchSnapshot();
  });

  describe(`Testing Simulate Render`, () => {
    it(`Testing Render On Desktop & Small Desktop Site`, async () => {
      const { findByTestId, getAllByTestId, getByTestId } = render(
        <MasonryLayout
          component={SampleComponent}
          item={MOCK_ITEM}
          maxGrid={4}
        />
      );

      /**
       * Simulate Resize Desktop Size
       */
      await act(simulateResize(1366));

      expect(await findByTestId(/masonry-container/)).toBeInTheDocument();
      expect(getByTestId(/masonry-container/)).toHaveStyle({
        'grid-template-columns': `repeat(4, 1fr)`
      });
      expect(getAllByTestId(/masonry-column/).length).toBe(4);

      /**
       * Simulate Resize Small Desktop Size
       */
      await act(simulateResize(1200));

      expect(await findByTestId(/masonry-container/)).toBeInTheDocument();
      expect(getByTestId(/masonry-container/)).toHaveStyle({
        'grid-template-columns': `repeat(4, 1fr)`
      });
      expect(getAllByTestId(/masonry-column/).length).toBe(4);
    });

    it(`Testing Render On Tablet & Mobile Site`, async () => {
      const { findByTestId, getAllByTestId, getByTestId } = render(
        <MasonryLayout
          component={SampleComponent}
          item={MOCK_ITEM}
          maxGrid={4}
        />
      );

      /**
       * Simulate Resize Tablet
       */
      await act(simulateResize(1024));

      expect(await findByTestId(/masonry-container/)).toBeInTheDocument();
      expect(getByTestId(/masonry-container/)).toHaveStyle({
        'grid-template-columns': `repeat(2, 1fr)`
      });
      expect(getAllByTestId(/masonry-column/).length).toBe(2);

      /**
       * Simulate Resize Small Desktop Size
       */
      await act(simulateResize(366));

      expect(await findByTestId(/masonry-container/)).toBeInTheDocument();
      expect(getByTestId(/masonry-container/)).toHaveStyle({
        'grid-template-columns': `repeat(1, 1fr)`
      });
      expect(getAllByTestId(/masonry-column/).length).toBe(1);
    });
  });
});
