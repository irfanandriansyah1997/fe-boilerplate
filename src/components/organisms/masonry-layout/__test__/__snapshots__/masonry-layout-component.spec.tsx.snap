// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Masonry Layout Componet Snapshot Testing 1`] = `
<div
  class="o-masonry"
  data-testid="masonry-container"
  style="grid-template-columns: repeat(4, 1fr);"
>
  <div
    data-testid="masonry-column"
  >
    <div
      data-testid="masonry-item"
    >
      Hello World
    </div>
    <div
      data-testid="masonry-item"
    >
      Hello World
    </div>
  </div>
  <div
    data-testid="masonry-column"
  >
    <div
      data-testid="masonry-item"
    >
      Hello World
    </div>
    <div
      data-testid="masonry-item"
    >
      Hello World
    </div>
  </div>
  <div
    data-testid="masonry-column"
  >
    <div
      data-testid="masonry-item"
    >
      Hello World
    </div>
    <div
      data-testid="masonry-item"
    >
      Hello World
    </div>
  </div>
  <div
    data-testid="masonry-column"
  >
    <div
      data-testid="masonry-item"
    >
      Hello World
    </div>
    <div
      data-testid="masonry-item"
    >
      Hello World
    </div>
  </div>
</div>
`;
